pygmt/src/filter1d.py

```diff
@@ -0,0 +1,144 @@
+"""
+filter1d - Time domain filtering of 1-D data tables
+"""
+import warnings
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile, build_arg_string, fmt_docstring, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    E="end",
+    F="filter",
```

Comment:
Following the alias in [GMT.jl](https://www.generic-mapping-tools.org/GMT.jl/dev/#GMT.filter1d):
```suggestion
    F="filter",
    N="time_col",
```